### DISCUSSION TITLE
How much percentage of otu (bacteria) are assigned to specific functional group (FAPROTAX)

### DISCUSSION THREAD

**Discussion by amrvx4** [ChiLiubio/microeco#243] [Q&A] [answered]
So, I got the data where I can see functional groups assigned based on FAPROTAX database. My question: is there any way to know specifically how much % of bacteria are assigned to specific group? For example, there was a group of CHEMOHETEROTROPHS based on FAPROTAX, but how do I know how much % of the bacteria/OTUs are assigned to that specific group?

Thanks,

**Comment by ChiLiubio** [comment, marked as answer]
Hi. The cal_spe_func_perc function with the parameter `abundance_weighted = FALSE` is used to get the % of OTUs/ASVs in each sample.
```
library(microeco)
data(dataset)
test <- trans_func$new(dataset)
test$cal_spe_func(prok_database = "FAPROTAX")
test$cal_spe_func_perc(abundance_weighted = FALSE)
View(test$res_spe_func_perc)
```

**Reply by Masoleno** [reply]
Hi! Following on from this question, is there a way to easily see which taxa (e.g. genus names) are assigned to which functional group?
Many thanks,
Maya

**Reply by ChiLiubio** [reply]
Sure. Here is an example.
```
library(microeco)
library(dplyr)

data(dataset)
test <- trans_func$new(dataset)
test$cal_spe_func(prok_database = "FAPROTAX")
View(test$res_spe_func)

# combine tables
combine_table <- cbind.data.frame(test$res_spe_func, dataset$tax_table[rownames(test$res_spe_func), "Genus", drop = FALSE])
# sum by genera
dedup_table <- combine_table %>%
    group_by(Genus) %>%
    summarise(across(everything(), sum))
```

**Reply by Masoleno** [reply]
Thank you so much from the bottom of my stressed-out-trying-to-finish-a-PhD heart!
